### ISSUE TITLE
instance norm batch size == 1

### ISSUE THREAD

**Issue by ddt2014** [arcchang1236/CA-NoiseGAN#3]
during inference, if set batch_size == 1, the instance_norm will raize valueerro: expected more than 1 value per channel, which is caused by the PyramidPooling's last 1x1 avg_pool2d. This layer resize the 'out' tensor  to [1, 32, 1, 1] shape, and this doesnot satisy the ‘_verify_batch_size()’ in pytorch instance_norm.
how to solve this, or only can infer image with batch_size > 1?

**Comment by arcchang1236** [comment]
If your PyTorch version isn't **1.1.0**? In this version, the batch size can be set to 1 to neglect the norm before pooling.
However, I strongly suggest using the **batch_size > 1** in inference time to get better performance.

**Comment by ddt2014** [comment]
As you use instance norm in netG, why batchsize>1 can lead to better performance?
And, during your training, whats ths batchsize for each gpu with instance norm in netG

**Comment by ddt2014** [comment]
And have you ever tried remove instancenorm or batchnorm after spectralnorm, as wgan loss may be perform better with spectralnorm.

**Comment by arcchang1236** [comment]
> Why batchsize>1 can lead to better performance?

- Sorry! I mean in training time but not in inference time and I edit the comment before.
- If you want to use batch_size=1 in inference time, duplicating the tensor by `torch.repeat()` is an alternative solution.

> During your training, what's the batch size for each GPU with instance norm in `netG`?

- I set 64 for training noise models and 128 for training denoising models

> Have you ever tried to remove instance norm or batch norm after spectral norm?

- Yes, since the unstable training of GAN, I use instance norm after spectral norm in our final model to get better performance. More, we found the training process can be more stable, too.
